androidApp/src/main/java/dev/johnoreilly/confetti/ui/ConfettiScaffold.kt

```diff
@@ -61,10 +65,22 @@ fun ConfettiScaffold(
     content: @Composable (ScaffoldState) -> Unit,
 ) {
     val authentication = koinInject<Authentication>()
+    val wearSettingSync = koinInject<WearSettingsSync>()
```

Comment:
Don't we create multiple of these, seem to be included in 4 different screens.  Would it be worth hoisting this to the top level VM?

```
fun ConfettiApp(
    navController: NavHostController,
    windowSizeClass: WindowSizeClass,
    displayFeatures: List<DisplayFeature>,
) {
    val appState: ConfettiAppState = rememberConfettiAppState(
        windowSizeClass,
        displayFeatures,
        navController
    )
```

So we only do this once for activity?

Reply:
I think that could be a good idea! Am I understanding this correct that it should be added as part of the AppState then right? Not passed through the composables? :)

Reply:
For me, yep.

Reply:
What about registering it as a singleton in DI? If we know we want only one, we could enforce it there instead of having an `AppState` that grows forever? Then we have Koin modules growing forever but we need to register the constructor there anyway, might as well make it `single {}`?

Reply:
Actually looks like [it's already the case](https://github.com/joreilly/Confetti/blob/afecf03f840941d0f04962c3a099084c755876fc/androidApp/src/main/java/dev/johnoreilly/confetti/di/AppModule.kt#L59)?